plugin/trino-example-plugin/pom.xml

```diff
@@ -0,0 +1,261 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
```

Comment:
Does this plugin get published as part of `mvn install` or `mvn release` (in addition to building the archetype)? If so, then this needs to be a group id under io.trino, or we won't be able to release to maven central.

Reply:
It's not supposed to be published.

Reply:
Should I remove it from the root `pom.xml`?

Reply:
If you remove it from the root pom, it won't get loaded by tools like IntelliJ

Reply:
Right. I'm not sure how releases are performed exactly, but it should be excluded during the release. https://github.com/trinodb/release-scripts doesn't seem to have anything related to the main artifacts.

Reply:
We do `mvn release:prepare` followed by `mvn release:perform`. It should also work with `mvn deploy`.

Reply:
Oh I know, I can configure the `maven-deploy-plugin` to be skipped in this module.

Reply:
I added an intermediate parent module for this new example plugin.

Reply:
Why do we need this extra parent?

Reply:
I also don't see why we're going through extra effort to avoid publishing this module. We publish all the modules in Trino, even if they're not externally useful. Why do we want to special case something here?

Reply:
The parent is needed so I can disable (skip) the deploy plugin inside it but without having this copied to the archetype. I probably could remove it in the archetype using some transformations (maven-antrun-plugin), but I wanted to avoid that. LMK which option would be preferable.

I also wanted to use a different groupId here because otherwise when building the archetype all `io.trino` imports would be messed up. Maybe we also could fix that using transformations.

Reply:
Ok that won't work. I'll remove the parent wrapper module, use a correct groupId and try to fix it when generating the archetype.

Reply:
Ok done, it wasn't that bad :-)